lib/components/Input/base/index.tsx

```diff
@@ -0,0 +1,135 @@
+import clsx from "clsx";
+import { InputHTMLAttributes, ReactNode, forwardRef, useState } from "react";
+import "./base.scss";
+import React from "react";
+import { TMediumSizes } from "../../../types";
+import { CaptionText } from "../../Typography";
+
+export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+    type?: "text" | "email" | "password";
+    icon?: ReactNode;
+    statusIcon?: ReactNode;
+    inputSize?: TMediumSizes;
+    status?: "neutral" | "success" | "error";
+    disabled?: boolean;
+    variant?: "fill" | "outline";
+    leftStatusMessage?: string;
+    rightStatusMessage?: string;
+    textAlignment?: "left" | "center";
+    label?: string;
+    value?: string;
+}
+
+const statusIconColors = {
+    neutral: "status-icon--neutral",
+    success: "status-icon--success",
+    error: "status-icon--error",
+};
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+    (
+        {
+            type = "text",
+            inputSize = "md",
+            className,
+            status = "neutral",
+            disabled = false,
+            variant = "outline",
+            icon: Icon,
+            leftStatusMessage,
+            rightStatusMessage,
+            textAlignment = "left",
+            label,
+            statusIcon: StatusIcon,
+            onChange,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [hasValue, setHasValue] = useState(false);
+
+        return (
+            <div className="quill-input__container">
+                <div
+                    className={clsx(
+                        className,
+                        `quill-input__wrapper`,
+                        hasValue && `quill-input__wrapper--has-value`,
+                        `quill-input__wrapper__variant--${variant}`,
+                        variant === "fill" && `status--${status}`,
+                        `quill-input__wrapper__size--${inputSize}`,
+                        `quill-input__wrapper__status--${status}`,
+                    )}
+                >
+                    {Icon && <span className="icon_wrapper">{Icon}</span>}
+                    <input
+                        {...rest}
+                        type={type}
+                        className={clsx(
+                            "input",
+                            "peer",
+                            `input__align--${textAlignment}`,
+                            `input__size--${inputSize}`,
+                        )}
+                        disabled={!!disabled}
+                        onChange={(e) => {
+                            setHasValue(!!e.target.value);
+                            onChange?.(e);
+                        }}
+                        ref={ref}
+                    />
+                    {label && inputSize === "md" && (
+                        <label
```

Comment:
Usually, it's a good practice to bind `label` with `input` by filling in `htmlFor` attribute (it can take `id` or `name` of the `input`). This is needed to increase accessibility of screen reader users (will read out loud the `label`, when the user is focused on the element) and users who have difficulty clicking on very small regions - because when a user clicks the text within the `<label>` element, it toggles the `input` (this increases the hit area).
But it's optional)))

Reply:
agreed on you @kate-deriv. in the figma given, label is within the input box, plus this component was migrated from Quill-design component which was having `pointer-events: none` for the label. To increase accessibility, will increase the height of label to be same with the input-wrapper and will bind the `input` and `label` with `htmlFor`.

Reply:
and will remove the `pointer-events: none` from the label.